XPlat/Shopify/iOS/source/Shopify.iOS/ApiDefinition.cs

```diff
@@ -0,0 +1,1964 @@
+
+		[Export ("imageHeightWithScrollViewDidScroll:")]
+		nfloat ImageHeightWithScrollViewDidScroll (UIScrollView scrollView);
+
+		[Export ("setCurrentPage:")]
+		void SetCurrentPage (nint currentPage);
+
+		[Export ("setImageForSelectedVariant:withImages:")]
+		void SetImageForSelectedVariant (ProductVariant productVariant, Image[] images);
+
+	}
+
+	[BaseType (typeof(UIView), Name = "BUYProductViewHeaderBackgroundImageView")]
+	interface ProductViewHeaderBackgroundImageView
+	{
+		[Export ("initWithTheme:")]
+		IntPtr Constructor (Theme theme);
+
+		[Export ("setBackgroundProductImage:")]
+		void SetBackgroundProductImage (Image image);
+
+	}
+
+	[BaseType (typeof(UIView), Name = "BUYProductViewHeaderOverlay")]
+	interface ProductViewHeaderOverlay
+	{
+		[Export ("initWithTheme:")]
+		IntPtr Constructor (Theme theme);
+
+		[Export ("scrollViewDidScroll:withNavigationBarHeight:")]
+		void ScrollViewDidScroll (UIScrollView scrollView, nfloat navigationBarHeight);
+
+	}
+
+	[BaseType (typeof(Object), Name = "BUYShop")]
+	interface Shop
+	{
+		[Export ("name")]
+		string Name { get; }
+
+		[Export ("city")]
+		string City { get; }
+
+		[Export ("province")]
+		string Province { get; }
+
+		[Export ("country")]
+		string Country { get; }
+
+		[Export ("currency")]
+		string Currency { get; }
+
+		[Export ("moneyFormat")]
+		string MoneyFormat { get; }
+
+		[Export ("domain")]
+		string Domain { get; }
+
+		[Export ("shopDescription")]
+		string ShopDescription { get; }
+
+		[Export ("shipsToCountries", ArgumentSemantic.Copy)]
+		string[] ShipsToCountries { get; }
+
+		[Export ("shopURL")]
+		NSUrl ShopUrl { get; }
+
+		[Export ("myShopifyURL")]
+		NSUrl MyShopifyUrl { get; }
+
+	}
+
+	delegate void CheckoutTypeBlock (CheckoutType checkoutType);
+
+	interface IStoreViewControllerDelegate
+	{
+
+	}
+
+	[Protocol, Model]
+	[BaseType(typeof(NSObject), Name = "BUYStoreViewControllerDelegate")]
+	interface StoreViewControllerDelegate : ViewControllerDelegate
+	{
+		[Abstract]
+		[Export ("controller:shouldProceedWithCheckoutType:")]
+		void ShouldProceedWithCheckoutType (StoreViewController controller, CheckoutTypeBlock completionHandler);
+
+	}
+
+	[BaseType (typeof(ViewController), Name = "BUYStoreViewController")]
+	interface StoreViewController
+	{
+		[Export ("initWithClient:url:")]
+		IntPtr Constructor (BuyClient client, NSUrl url);
+
+		[Export ("reloadHomePage")]
+		void ReloadHomePage ();
+
+		[NullAllowed, Export ("delegate", ArgumentSemantic.Weak)]
+		IStoreViewControllerDelegate Delegate { get; set; }
+
+	}
+
+	[BaseType (typeof(Object), Name = "BUYTaxLine")]
+	interface TaxLine
+	{
+		[Export ("price", ArgumentSemantic.Strong)]
+		NSDecimalNumber Price { get; set; }
+
+		[Export ("rate", ArgumentSemantic.Strong)]
+		NSDecimalNumber Rate { get; set; }
+
+		[Export ("title")]
+		string Title { get; set; }
+
+	}
+
+	[Category]
+	[BaseType (typeof(Theme), Name = "BUYTheme")]
+	interface ThemeExtensions
```

Comment:
As suggestion, I would change these methods to properties or, if you decide to leave them as methods, add a Get prefix.